src/main/java/net/tiagofar78/prisonescape/Events.java

```diff
@@ -67,9 +67,17 @@ public void playerMove(PlayerMoveEvent e) {
             return;
         }
 
+        Player player = e.getPlayer();
+        Block blockWalkedOn = player.getLocation().subtract(0, 1, 0).getBlock();
+        Material blockType = blockWalkedOn.getType();
         PrisonEscapeLocation location = new PrisonEscapeLocation(x, y, z);
 
-        game.playerMove(e.getPlayer().getName(), location);
+        if (blockType == Material.BLACK_STAINED_GLASS) {
```

Comment:
This would be a problem if a player is walking in black stained glass that is not part of a metal detector. And players wound be able to skip metal detectors if they jumped when going through them. You should just use locations to detect if a player is on a metal detector.

Another problem is that PrisonEscapeGame#playerMove function should be general enough to take care of everything related with movement.

Reply:
Oh okay didn't think about that scenario